lte/gateway/c/oai/tasks/s1ap/s1ap_mme_handlers.c

```diff
@@ -362,13 +362,20 @@ void clean_stale_enb_state(
     ue_description_t* ue_ref = NULL;
     for (int i = 0; i < keys->num_keys; i++) {
       ue_ref = s1ap_state_get_ue_mmeid((mme_ue_s1ap_id_t) keys->keys[i]);
+      /* The function s1ap_remove_ue will take care of removing the enb also,
```

Comment:
@VinashakAnkitAman , `s1ap_remove_ue` removes the eNB only if the enb->num_ue_associated is zero, but we have seen scenarios (with SCTP reset) where the `num_ue_associated` is non-zero even when all the UEs in ue_id_coll are removed (See https://github.com/magma/magma/pull/4899) .
A better check would be to set the enb_ref to NULL inside `s1ap_remove_enb` so that the second call to `s1ap_remove_enb` is a no-op.

Reply:
I am told this is resolved. Resolving to enable progress.